src/libsync/ocsprofileconnector.cpp

```diff
@@ -0,0 +1,148 @@
+#include "ocsprofileconnector.h"
+#include "common/result.h"
+#include "networkjobs.h"
+#include "iconjob.h"
+#include "theme.h"
+
+#include <QJsonObject>
+#include <QJsonDocument>
+#include <QJsonArray>
+#include <QLoggingCategory>
+#include <QIcon>
+#include <QPainter>
+#include <QImage>
+#include <QSvgRenderer>
+#include <qpixmap.h>
+
+namespace {
+Q_LOGGING_CATEGORY(lcOcsProfileConnector, "nextcloud.gui.ocsprofileconnector", QtInfoMsg)
+
+OCC::HovercardAction jsonToAction(const QJsonObject &jsonActionObject)
+{
+    return {jsonActionObject.value("title").toString("No title"), jsonActionObject.value("icon").toString("no-icon"),
```

Comment:
@FlexW You could as well have put strings within `value("")` to `QStringLiteral`. This ensures that the `QString` is created in compile-time.